### ISSUE TITLE
fast_tsne returns NaN for all x and y

### ISSUE THREAD

**Issue by fbecerra** [KlugerLab/FIt-SNE#76]
I've been using the compiled version of fast_tsne for a while, but recently I updated and recompiled it and realized that it's not working for me anymore. Commit bd10ce4 is the problematic one, it basically saves only NaN in result.dat and the cost in the output is also NaN:

> ============== t-SNE v1.0.1 ===============
fast_tsne data_path: data.dat
fast_tsne result_path: result.dat
fast_tsne nthreads: 4
Read the following parameters:
	 n 5989 by d 50 dataset, theta 0.500000,
	 perplexity 30.000000, no_dims 2, max_iter 1000,
	 stop_lying_iter 250, mom_switch_iter 250,
	 momentum 0.500000, final_momentum 0.800000,
	 learning_rate 200.000000, K -1, sigma -1.000000, nbody_algo 2,
	 knn_algo 1, early_exag_coeff 12.000000,
	 no_momentum_during_exag 0, n_trees 50, search_k 4500,
	 start_late_exag_iter -1, late_exag_coeff -1.000000
	 nterms 3, interval_per_integer 1.000000, min_num_intervals 50, t-dist df 0.000000
Read the 5989 x 50 data matrix successfully. X[0,0] = 0.335302
Will use momentum during exaggeration phase
Computing input similarities...
Using perplexity, so normalizing input data (to prevent numerical problems)
Using perplexity, not the manually set kernel width.  K (number of nearest neighbors) and sigma (bandwidth) parameters are going to be ignored.
Using ANNOY for knn search, with parameters: n_trees 50 and search_k 4500
Going to allocate memory. N: 5989, K: 90, N*K = 539010
Building Annoy tree...
Done building tree. Beginning nearest neighbor search...
parallel (4 threads):
[===========================================================>] 99% 0.844s
Symmetrizing...
Using current time as random seed...
Randomly initializing the solution.
Y[0] = -0.000211
Exaggerating Ps by 12.000000
Input similarities computed (sparsity = 0.023972)!
Learning embedding...
Using FIt-SNE approximation.
Iteration 50 (50 iterations in 0.80 seconds), cost nan
Iteration 100 (50 iterations in 0.77 seconds), cost nan
Iteration 150 (50 iterations in 0.78 seconds), cost nan
Iteration 200 (50 iterations in 0.77 seconds), cost nan
Iteration 250 (50 iterations in 0.77 seconds), cost nan
Unexaggerating Ps by 12.000000
Iteration 300 (50 iterations in 0.78 seconds), cost nan
Iteration 350 (50 iterations in 0.77 seconds), cost nan
Iteration 400 (50 iterations in 0.78 seconds), cost nan
Iteration 450 (50 iterations in 0.78 seconds), cost nan
Iteration 500 (50 iterations in 0.77 seconds), cost nan
Iteration 550 (50 iterations in 0.78 seconds), cost nan
Iteration 600 (50 iterations in 0.77 seconds), cost nan
Iteration 650 (50 iterations in 0.86 seconds), cost nan
Iteration 700 (50 iterations in 0.80 seconds), cost nan
Iteration 750 (50 iterations in 0.85 seconds), cost nan
Iteration 800 (50 iterations in 0.81 seconds), cost nan
Iteration 850 (50 iterations in 0.92 seconds), cost nan
Iteration 900 (50 iterations in 0.81 seconds), cost nan
Iteration 950 (50 iterations in 0.90 seconds), cost nan
Iteration 1000 (50 iterations in 0.78 seconds), cost nan
Wrote the 5989 x 2 data matrix successfully.
Done.

Before I was using commit 4dd27e0 and everything wsa working fine, with exactly the same data.dat file:

> =============== t-SNE ===============
fast_tsne data_path: data.dat
fast_tsne result_path: result.dat
fast_tsne nthreads: 4
Read the following parameters:
	 n 5989 by d 50 dataset, theta 0.500000,
	 perplexity 30.000000, no_dims 2, max_iter 1000,
	 stop_lying_iter 250, mom_switch_iter 250,
	 momentum 0.500000, final_momentum 0.800000,
	 learning_rate 200.000000, K -1, sigma -1.000000, nbody_algo 2,
	 knn_algo 1, early_exag_coeff 12.000000,
	 no_momentum_during_exag 0, n_trees 50, search_k 4500,
	 start_late_exag_iter -1, late_exag_coeff -1.000000
	 nterms 3, interval_per_integer 1.000000, min_num_intervals 50
Read the 5989 x 50 data matrix successfully. X[0,0] = 0.335302
Using current time as random seed...
Will use momentum during exaggeration phase
Computing input similarities...
Using perplexity, so normalizing input data (to prevent numerical problems)
Using perplexity, not the manually set kernel width.  K (number of nearest neighbors) and sigma (bandwidth) parameters are going to be ignored.
Using ANNOY for knn search, with parameters: n_trees 50 and search_k 4500
Going to allocate memory. N: 5989, K: 90, N*K = 539010
Building Annoy tree...
Done building tree. Beginning nearest neighbor search...
parallel (4 threads):
[============================================================] 100% 0.864s
Symmetrizing...
Randomly initializing the solution.
Exaggerating Ps by 12.000000
Input similarities computed (sparsity = 0.023972)!
Learning embedding...
Using FIt-SNE approximation.
Iteration 50 (50 iterations in 0.72 seconds), cost 4.998925
Iteration 100 (50 iterations in 0.77 seconds), cost 4.659932
Iteration 150 (50 iterations in 0.82 seconds), cost 4.645344
Iteration 200 (50 iterations in 0.76 seconds), cost 4.636345
Iteration 250 (50 iterations in 0.76 seconds), cost 4.622303
Unexaggerating Ps by 12.000000
Iteration 300 (50 iterations in 0.74 seconds), cost 2.721241
Iteration 350 (50 iterations in 0.81 seconds), cost 2.308060
Iteration 400 (50 iterations in 0.94 seconds), cost 2.093742
Iteration 450 (50 iterations in 0.90 seconds), cost 1.965181
Iteration 500 (50 iterations in 1.33 seconds), cost 1.875860
Iteration 550 (50 iterations in 1.68 seconds), cost 1.809087
Iteration 600 (50 iterations in 2.08 seconds), cost 1.765513
Iteration 650 (50 iterations in 2.71 seconds), cost 1.729019
Iteration 700 (50 iterations in 2.68 seconds), cost 1.708275
Iteration 750 (50 iterations in 2.96 seconds), cost 1.693048
Iteration 800 (50 iterations in 3.96 seconds), cost 1.681363
Iteration 850 (50 iterations in 4.81 seconds), cost 1.673190
Iteration 900 (50 iterations in 4.28 seconds), cost 1.660609
Iteration 950 (50 iterations in 4.76 seconds), cost 1.652336
Iteration 1000 (50 iterations in 6.39 seconds), cost 1.645526
Wrote the 5989 x 2 data matrix successfully.
Done.

**Comment by dkobak** [comment]
> ============== t-SNE v1.0.1 ===============

The current version is 1.1.0 and this is what you should see in this output (check https://github.com/KlugerLab/FIt-SNE/blob/master/src/tsne.cpp#L1875). Please make sure to `git pull` the latest version,  recompile, and double-check that that's what you are using.

**Comment by dkobak** [comment]
Also, I am not sure you can use the same `data.dat` as with previous versions because its format might have changed. You need to run your code again, so that the wrapper can generate new `data.dat`.

**Comment by fbecerra** [comment]
Thanks! I was using an older version of the fast_tsne.py that was causing this issue. I pulled the latest version and it seems to be working fine. Thanks again.
